SECP-service/src/main/java/com/visucius/secp/resources/UserResource.java

```diff
@@ -23,19 +26,31 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(UserResource.class);
 
+
     private final UserDAO dao;
+    private final UserRegistrationController userRegistrationController;
+
+    public UserResource(UserDAO dao, UserRegistrationController userRegistrationController) {
 
-    public UserResource(UserDAO dao) {
         this.dao = dao;
+        this.userRegistrationController = userRegistrationController;
     }
 
     @POST
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
-    @Timed 
+    @Timed
     @UnitOfWork
-    public User create(User entity) {
-        return dao.save(entity);
+    @Path("/register")
+    public UserRegistrationResponse create(UserRegistrationRequest request) {
+
+        UserRegistrationResponse response = userRegistrationController.handle(request);
```

Comment:
You don't need to create a response class. There is already a ResponseBuilder class for the project. Look at https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/Response.ResponseBuilder.html for more information,